src/utilities/helperFunctions.js

```diff
@@ -0,0 +1,15 @@
+// eslint-disable-next-line import/prefer-default-export
+export const validate = (input, setErrorMessage) => {
+  let validated = false;
+  const regex = /^[\p{L}(0-9)]+(?:[\s-][\p{L}(0-9)]+)*$/u;
```

Comment:
This regex checks if the string input consist of one or more string characters or numbers. A string could pass this regex validation but not be a valid city name. For example hello123 would pass the regex and the validation (results tested using [regex101](https://regex101.com/). Therefore, always this checks if the user inputs something, I think the error messages are incorrect. I think you can make this function check if the user input is a string, then the city name validation can be done where the geocoding API call is being made (which is already implemented).

Reply:
I thought the same thing too, but then I also saw there were cities that contained digits, like for example, Mila 23 in Romania.

But I also could have clarified exactly what the regex is checking for in a comment so adjustments to it can be made as needed.

Reply:
Oh really, I didn't know that. If that's the case, the regex could technically stay. Its tricky because regexes are used to check for a specific pattern in a string and it isn't possible to check if the string is a valid city name with regexes alone. However, it is useful for if the user inputs just a number or special character.

Reply:
Yeah, I think a more robust validation would check against the OpenWeather city database, which I think would be doable! If you think that's a good idea I'd like to separate that into another issue since it feels like more of an enhancement to me.

Reply:
Of course, validating city name could be a new issue and an enhancement if we'd like to add.